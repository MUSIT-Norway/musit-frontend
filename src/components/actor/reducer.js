import { Observable } from 'rxjs/Rx';
import actions from "./actions";

const dataFromServer = [
  {
    id: 1,
    fn: 'Jarl parl'
  },
  {
    id: 2,
    fn: 'Line pine'
  },
  {
    id: 3,
    fn: 'Kjell pell'
  },
  {
    id: 4,
    fn: 'Stein sein'
  },
  {
    id: 5,
    fn: 'Knut lut'
  },
  {
    id: 6,
    fn: 'Ellen'
  }
]

const initialState = [];

const ActorReducer$ = Observable.of(() => initialState)
  .merge(
    actions.suggest$.map((update) => () => {
      if (!update) {
        return initialState;
      }
      return dataFromServer.filter(d => {
        const name = d.fn.toLowerCase();
        const query = update.value.toLowerCase();
        return name.indexOf(query) > -1
      })
    }),
    actions.clear$.map(() => () => initialState),
  );

export default Observable.merge(
  ActorReducer$.map(reducer => ["suggest", reducer]),
);