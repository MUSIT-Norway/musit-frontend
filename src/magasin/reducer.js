import { toFrontend } from './mapper';
import * as types from './types';

const initialState = {
  root: {
    data: {}
  }
};

const storageUnitGridReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case types.LOAD_SEVERAL:
    return {
      ...state,
      loading: true
    };
  case types.LOAD_SEVERAL_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.result
    };
  case types.LOAD_SEVERAL_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  case types.LOAD_ONE_SUCCESS:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: true,
        data: toFrontend(action.result)
      }
    };
  case types.LOAD_ONE_FAIL:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: false,
        error: action.error
      }
    };
  case types.CLEAR_ROOT: {
    return {
      ...state,
      root: {}
    };
  }
  case types.DELETE:
    return {
      ...state,
      loading: true
    };
  case types.DELETE_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: []
    };
  case types.DELETE_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  default:
    return state;
  }
};

export default storageUnitGridReducer;