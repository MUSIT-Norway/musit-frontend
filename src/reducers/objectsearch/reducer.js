import { omit } from 'lodash';
import { getPath } from '../helper';
import MusitObject from '../../models/object';
import * as types from './constants';

const initialState = {
  data: {
    totalMatches: 0,
    matches: []
  },
  params: {
    currentPage: 1,
    perPage: 50
  },
  loaded: true
};

export default (state = initialState, action) => {
  switch(action.type) {
  case types.CHANGE_FIELD:
    return {
      ...state,
      params: {
        ...state.params,
        [action.field]: action.value
      }
    };
  case types.SEARCH_OBJECTS:
    return {
      ...omit(state, 'error'),
      loading: true,
      loaded: false,
      data: {
        totalMatches: 0,
        matches: []
      },
      params: {
        ...state.params,
        currentPage: action.page
      }
    };
  case types.SEARCH_OBJECTS_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      data: [],
      error: action.error
    };
  case types.SEARCH_OBJECTS_SUCCESS:
    return {
      ...omit(state, 'error'),
      loading: false,
      loaded: true,
      data: {
        totalMatches: action.result.totalMatches,
        matches: action.result.matches.map(data => {
          return new MusitObject({
            ...data,
            breadcrumb: getPath(data)
          });
        })
      }
    };
  default:
    return state;
  }
};