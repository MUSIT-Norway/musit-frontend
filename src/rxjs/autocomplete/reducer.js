import { Observable } from 'rxjs/Rx';

export default function createReducer(actions) {
  const initialState = [];
  return Observable.of(() => initialState)
    .merge(
      actions.update$.map((data) => () => data),
      actions.clear$.map(() => () => initialState),
    );
}