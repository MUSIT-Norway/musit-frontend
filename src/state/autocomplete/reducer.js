import { Observable } from 'rxjs/Rx';

export default function createReducer(actions, stateKey = "suggest") {
  const initialState = [];
  const reducer$ = Observable.of(() => initialState)
    .merge(
      actions.update$.map((data) => () => data),
      actions.clear$.map(() => () => initialState),
    );
  return Observable.merge(
    reducer$.map(reducer => [stateKey, reducer]),
  );
}