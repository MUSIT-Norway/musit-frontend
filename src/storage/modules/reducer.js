import { toFrontend } from '../../magasin/modules/mapper';
import * as types from './types';

const initialState = {
  state: {
    environmentRequirement: {},
    environmentAssessment: {},
    securityAssessment: {}
  }
};

const storageUnitContainerReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case types.INSERT:
    return {
      ...state,
      loading: true
    };
  case types.INSERT_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: toFrontend(action.result)
    };
  case types.INSERT_FAIL:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.error
    };
  case types.UPDATE:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.data
    };
  case types.LOAD:
    return {
      ...state,
      loading: true
    };
  case types.LOAD_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: toFrontend(action.result)
    };
  case types.LOAD_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  case types.UPDATE_STATE:
    return {
      ...state,
      state: action.data
    };
  case types.CLEAR_STATE:
    return {
      ...state,
      state: {
        environmentRequirement: {},
        environmentAssessment: {},
        securityAssessment: {}
      }
    };
  default:
    return state;
  }
};

export default storageUnitContainerReducer;