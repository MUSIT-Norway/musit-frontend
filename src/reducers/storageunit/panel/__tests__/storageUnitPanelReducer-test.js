
import * as actions from '../index';
import reducer from '../index';
import Config from '../../../../config';
import request from 'superagent';
import nocker from 'superagent-nock';
const nock = nocker(request);


describe('ReducerStoragUnitPanel', () => {
  const loadState = {
    id: 3,
    name: 'Forskningens hus',
    isPartOf: 2,
    groupRead: 'foo',
    path: ',1,2,3,',
    pathNames: [
      {
        nodeId: 1,
        name: 'root-node'
      },
      {
        nodeId: 2,
        name: 'Utviklingsmuseet'
      },
      {
        nodeId: 3,
        name: 'Forskningens hus'
      }],
    environmentRequirement: {
      comment: "fdsfd"
    },
    updatedBy:666,
    updatedDate: '2016-10-24T15:41:35+00:00',
    type: 'Building'
  };
  it('creates LOAD_SUCCESS when fetching data has been done', () => {
    const id = 3;
    const url = `${Config.magasin.urls.storagefacility.baseUrl(1)}/${id}`;
    nock('http://localhost')
            .get(url)
            .reply(200, loadState);
    const store = mockStore();

    return store.dispatch(actions.load(3))
            .then(() => {
              expect(store.getActions()).toMatchSnapshot();
            });
  });

  it('LOAD: no action', () => {
    expect(
            reducer(undefined, {})
        ).toMatchSnapshot();
  });

  it('LOAD: initial action', () => {
    expect(
            reducer(undefined, {
              type: actions.LOAD
            })
        ).toMatchSnapshot();
  });

  it('LOAD: success action', () => {
    expect(
            reducer(undefined, {
              type: actions.LOAD_SUCCESS,
              result: {
                someField: 1
              }
            })
        ).toMatchSnapshot();
  });

  it('LOAD: fail action', () => {
    expect(
            reducer(undefined, {
              type: actions.LOAD_FAIL,
              error: Error('LOAD has some error.')
            })
        ).toMatchSnapshot();
  });
  const putData = {
    id: 2,
    name: "p5",
    isPartOf:1,
    groupRead: 'foo',
    path: ',1,2,',
    pathNames:[
      {
        nodeId: 1,
        name: 'root-node'
      },
      {
        nodeId: 2,
        name: 'Utviklingsmuseet'
      }
    ],
    environmentRequirement: {},
    securityAssessment: {},
    environmentAssessment: {},
    updatedBy: 123,
    updatedDate: '2016-10-24T16:13:24+00:00',
    type: 'Room'
  };
  it('update INSERT_SUCCESS when fetching data has been done', () => {
    const id = 2;
    const url = `${Config.magasin.urls.storagefacility.baseUrl(1)}/${id}`;
    nock('http://localhost')
            .put(url, putData)
            .reply(201, putData);
    const store = mockStore();

    return store.dispatch(actions.update(putData))
            .then(() => {
              expect(store.getActions()).toMatchSnapshot();
            });
  });


  it('update: no action', () => {
    expect(
            reducer(undefined, {})
        ).toMatchSnapshot();
  });

  it('update: initial action', () => {
    expect(
            reducer(undefined, {
              type: actions.INSERT
            })
        ).toMatchSnapshot();
  });

  it('update: success action', () => {
    expect(
            reducer(undefined, {
              type: actions.INSERT_SUCCESS,
              result: {
                someField: 1
              }
            })
        ).toMatchSnapshot();
  });

  it('update: fail action', () => {
    expect(
            reducer(undefined, {
              type: actions.INSERT_FAIL,
              error: Error('Some error in update.')
            })
        ).toMatchSnapshot();
  });

  it('insert INSERT_SUCCESS when fetching data has been done', () => {
    const postData = {
      name: "p5",
      isPartOf:1,
      groupRead: 'foo',
      environmentRequirement: {},
      securityAssessment: {},
      environmentAssessment: {},
      type: 'Room',
      area: 1
    };
    const parentId = 1;
    const url = `${Config.magasin.urls.storagefacility.baseUrl(1)}${!parentId ? '/root' : ''}`;
    nock('http://localhost')
            .post(url, postData)
            .reply(201, putData);
    const store = mockStore();

    return store.dispatch(actions.insert(1, postData))
            .then(() => {
              expect(store.getActions()).toMatchSnapshot();
            });
  });


  it('insert: no action', () => {
    expect(
            reducer(undefined, {})
        ).toMatchSnapshot();
  });

  it('insert: initial action', () => {
    expect(
            reducer(undefined, {
              type: actions.INSERT
            })
        ).toMatchSnapshot();
  });

  it('insert: success action', () => {
    expect(
            reducer(undefined, {
              type: actions.INSERT_SUCCESS,
              result: {
                someField: 1
              }
            })
        ).toMatchSnapshot();
  });

  it('insert: fail action', () => {
    expect(
            reducer(undefined, {
              type: actions.INSERT_FAIL,
              error: Error('Some error in insert.')
            })
        ).toMatchSnapshot();
  });
});
