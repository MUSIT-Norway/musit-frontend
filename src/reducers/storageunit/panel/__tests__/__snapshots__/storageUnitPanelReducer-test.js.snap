exports[`ReducerStoragUnitPanel LOAD: fail action 1`] = `
Object {
  "error": [Error: LOAD has some error.],
  "loaded": false,
  "loading": false
}
`;

exports[`ReducerStoragUnitPanel LOAD: initial action 1`] = `
Object {
  "loading": true
}
`;

exports[`ReducerStoragUnitPanel LOAD: no action 1`] = `Object {}`;

exports[`ReducerStoragUnitPanel creates LOAD_SUCCESS when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/storageunit-container/LOAD"
  },
  Object {
    "result": Object {
      "environmentRequirement": Object {
        "comment": "fdsfd"
      },
      "groupRead": "foo",
      "id": 3,
      "isPartOf": 2,
      "name": "Forskningens hus",
      "path": ",1,2,3,",
      "pathNames": Array [
        Object {
          "name": "root-node",
          "nodeId": 1
        },
        Object {
          "name": "Utviklingsmuseet",
          "nodeId": 2
        },
        Object {
          "name": "Forskningens hus",
          "nodeId": 3
        }
      ],
      "type": "Building",
      "updatedBy": 666,
      "updatedDate": "2016-10-24T15:41:35+00:00"
    },
    "type": "musit/storageunit-container/LOAD_SUCCESS"
  }
]
`;

exports[`ReducerStoragUnitPanel insert INSERT_SUCCESS when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/storageunit-container/INSERT"
  },
  Object {
    "result": Object {
      "environmentAssessment": Object {},
      "environmentRequirement": Object {},
      "groupRead": "foo",
      "id": 2,
      "isPartOf": 1,
      "name": "p5",
      "path": ",1,2,",
      "pathNames": Array [
        Object {
          "name": "root-node",
          "nodeId": 1
        },
        Object {
          "name": "Utviklingsmuseet",
          "nodeId": 2
        }
      ],
      "securityAssessment": Object {},
      "type": "Room",
      "updatedBy": 123,
      "updatedDate": "2016-10-24T16:13:24+00:00"
    },
    "type": "musit/storageunit-container/INSERT_SUCCESS"
  }
]
`;

exports[`ReducerStoragUnitPanel insert: fail action 1`] = `
Object {
  "data": [Error: Some error in insert.],
  "loaded": false,
  "loading": false
}
`;

exports[`ReducerStoragUnitPanel insert: initial action 1`] = `
Object {
  "loading": true
}
`;

exports[`ReducerStoragUnitPanel insert: no action 1`] = `Object {}`;

exports[`ReducerStoragUnitPanel update INSERT_SUCCESS when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/storageunit-container/INSERT"
  },
  Object {
    "result": Object {
      "environmentAssessment": Object {},
      "environmentRequirement": Object {},
      "groupRead": "foo",
      "id": 2,
      "isPartOf": 1,
      "name": "p5",
      "path": ",1,2,",
      "pathNames": Array [
        Object {
          "name": "root-node",
          "nodeId": 1
        },
        Object {
          "name": "Utviklingsmuseet",
          "nodeId": 2
        }
      ],
      "securityAssessment": Object {},
      "type": "Room",
      "updatedBy": 123,
      "updatedDate": "2016-10-24T16:13:24+00:00"
    },
    "type": "musit/storageunit-container/INSERT_SUCCESS"
  }
]
`;

exports[`ReducerStoragUnitPanel update: fail action 1`] = `
Object {
  "data": [Error: Some error in update.],
  "loaded": false,
  "loading": false
}
`;

exports[`ReducerStoragUnitPanel update: initial action 1`] = `
Object {
  "loading": true
}
`;

exports[`ReducerStoragUnitPanel update: no action 1`] = `Object {}`;
